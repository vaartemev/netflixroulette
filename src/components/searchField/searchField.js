import React,{ useState } from 'react';
import { SearchControls } from '../searchControls/searchControls';
import { Input } from '../input/input';

import './searchField.scss';

export const SearchField = () => {
  const [searchValue, setValue] = useState('');
  return (
    <div className="search">
      <div className="search__title">Find your movie</div>
      <Input searchValue={searchValue} setValue={setValue}/>
      <SearchControls searchValue={searchValue}/>
    </div>
  );
};
