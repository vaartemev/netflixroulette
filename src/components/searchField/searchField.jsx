import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { SearchControls } from '../../containers/searchControls/searchControls';
import { Input } from '../input/input';

import './searchField.scss';

export const SearchField = ({ handleOnEnterPress }) => {
  const [searchValue, setValue] = useState('');
  return (
    <div className="search">
      <div className="search__title">Find your movie</div>
      <Input
        searchValue={searchValue}
        setValue={setValue}
        handleOnEnterPress={handleOnEnterPress}
        placeholderText="What do you what find?"
      />
      <SearchControls searchValue={searchValue} />
    </div>
  );
};

SearchField.propTypes = {
  handleOnEnterPress: PropTypes.func
};
