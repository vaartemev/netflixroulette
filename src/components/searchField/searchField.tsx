import React from 'react';
import { SearchControls } from './searchControls';
import { Input } from '../input';

import './searchField.scss';

interface Props {
  handleOnEnterPress: () => void;
  handleOnInput: () => void;
  handleOnSearchClick: () => void;
  searchValue: string | string[];
  searchFilter: string;
  sortFilter: string;
}

export const SearchField = ({
  handleOnEnterPress,
  handleOnInput,
  handleOnSearchClick,
  searchValue,
  searchFilter,
  sortFilter,
}: Props) => {
  return (
    <div className="search">
      <div className="search__title">Find your movie</div>
      <Input
        searchValue={searchValue}
        handleOnInput={handleOnInput}
        handleOnEnterPress={handleOnEnterPress}
        placeholderText="What do you what find?"
      />
      <SearchControls
        searchValue={searchValue}
        searchFilter={searchFilter}
        sortFilter={sortFilter}
        handleOnSearchClick={handleOnSearchClick}
      />
    </div>
  );
};
