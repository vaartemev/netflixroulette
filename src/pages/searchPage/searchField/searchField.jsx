import React from 'react';
import PropTypes from 'prop-types';
import { SearchControls } from '../../../components/searchControls/searchControls';
import { Input } from '../../../components/input/input';

import './searchField.scss';

export const SearchField = ({
  handleOnEnterPress,
  handleOnInput,
  handleOnSearchClick,
  searchValue,
  searchFilter,
  sortFilter,
}) => {
  return (
    <div className="search">
      <div className="search__title">Find your movie</div>
      <Input
        searchValue={searchValue}
        handleOnInput={handleOnInput}
        handleOnEnterPress={handleOnEnterPress}
        placeholderText="What do you what find?"
      />
      <SearchControls
        searchValue={searchValue}
        searchFilter={searchFilter}
        sortFilter={sortFilter}
        handleOnSearchClick={handleOnSearchClick}
      />
    </div>
  );
};

SearchField.propTypes = {
  handleOnEnterPress: PropTypes.func,
  handleOnInput: PropTypes.func,
  handleOnSearchClick: PropTypes.func,
  searchValue: PropTypes.string,
  searchFilter: PropTypes.string,
  sortFilter: PropTypes.string,
};
