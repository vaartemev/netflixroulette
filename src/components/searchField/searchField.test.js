import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { SearchField } from './searchField';

describe('Search Field', () => {
  const props = {
    searchValue: 'string2',
    searchFilter: 'string',
    sortFilter: 'string2',
    handleOnSearchClick: f => f(),
    handleOnEnterPress: f => f(),
    handleOnInput: f => f()
  };

  const output = shallow(<SearchField {...props} />);

  it('renders correctly', () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
