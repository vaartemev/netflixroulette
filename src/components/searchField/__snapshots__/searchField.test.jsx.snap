// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Field renders correctly 1`] = `
<div
  className="search"
>
  <div
    className="search__title"
  >
    Find your movie
  </div>
  <Input
    handleOnEnterPress={[Function]}
    handleOnInput={[Function]}
    placeholderText="What do you what find?"
    searchValue="string2"
  />
  <SearchControls
    handleOnSearchClick={[Function]}
    searchFilter="string"
    searchValue="string2"
    sortFilter="string2"
  />
</div>
`;
